src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -203,225 +167,119 @@ public HttpResponseMessage RotateDocumentPages(PostedDataEntity postedData)
+                var pageNumber = postedData.pages[0];
+
+                string pageFilePathFormat;
+                ViewerSettings settings = GetViewerSettings(documentGuid, out pageFilePathFormat);
+
+                var cacheFolder = ((FileCache)settings.Cache).CachePath;
+                File.Delete(Path.Combine(cacheFolder, "p" + pageNumber + "_html.dat"));
+
+                using (GroupDocs.Viewer.Viewer viewer = new GroupDocs.Viewer.Viewer(documentGuid, settings))
                 {
-                    // prepare rotated page info object
-                    RotatedPageEntity rotatedPage = new RotatedPageEntity();
-                    int pageNumber = pages[i];
-                    RotatePageOptions rotateOptions = new RotatePageOptions(pageNumber, angle);
-                    // perform page rotation
-                    string resultAngle = "0";
-                    // set password for protected document
-                    if (!string.IsNullOrEmpty(password))
-                    {
-                        rotateOptions.Password = password;
-                    }
-                    this.GetHandler().RotatePage(documentGuid, rotateOptions);
-                    resultAngle = this.GetHandler().GetDocumentInfo(documentGuid, documentInfoOptions).Pages[pageNumber - 1].Angle.ToString();
-                    // add rotated page number
-                    rotatedPage.SetPageNumber(pageNumber);
-                    // add rotated page angle
-                    rotatedPage.SetAngle(resultAngle);
-                    // add rotated page object into resulting list                   
-                    rotatedPages.Add(rotatedPage);
+                    HtmlViewOptions viewOptions = HtmlViewOptions.ForEmbeddedResources();
+                    Rotation angle = GetAngleValue(postedData.angle);
+
+                    viewOptions.RotatePage(1, angle);
+                    viewer.View(viewOptions);
                 }
-                return Request.CreateResponse(HttpStatusCode.OK, rotatedPages);
+
+                // TODO: remove response object
+                return Request.CreateResponse(HttpStatusCode.OK, new object());
             }
-            catch (System.Exception ex)
+            catch (Exception ex)
             {
                 // set exception message
-                return Request.CreateResponse(HttpStatusCode.InternalServerError, new Resources().GenerateException(ex));
+                return Request.CreateResponse(HttpStatusCode.OK, new Resources().GenerateException(ex));
             }
         }
 
-        /// <summary>
-        /// Download curerntly viewed document
-        /// </summary>
-        /// <param name="path">Path of the document to download</param>
-        /// <returns>Document stream as attachement</returns>
-        [HttpGet]
-        [Route("downloadDocument")]
-        public HttpResponseMessage DownloadDocument(string path)
+        private Rotation GetAngleValue(int angle)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetAngleValue' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)